substrate/pkg/controller/substrate/infrastructure/subnets.go

```diff
@@ -46,6 +46,9 @@ func (s *Subnets) Create(ctx context.Context, substrate *v1alpha1.Substrate) (re
 		return reconcile.Result{}, err
 	}
 	for _, subnet := range subnets {
+		if subnet == nil { // we can run into a case when ctx is canceled, errs and subnets are all nil
```

Comment:
If ctx is cancelled, don't you have an error value caught on line 45?

Reply:
No I ran into this, err was nil and because we have subnets slice with nils defined on line 40, we enter this for loop.

Reply:
There's gotta be a bug somewhere else.